### PR TITLE
Chat settings redesign

### PR DESCRIPTION
**Pull request by DispatchCommit**
Creates a categorized, tabbed version of chat settings to accommodate more settings.
Separates large single settings file into 3 smaller sub components, one for each settings panel.
Settings overflow within the card, rather than entire card.

Changes TTS voice selection box to solo variant.
Moves all toggles to be in line.
Increase menu title bar font weight to bold.

Current Design:
image
image
image

### CODE REVIEW

File: components/Chat/ChatMessages/ChatMessageMenu.vue [hunk 1]
```diff
@@ -107,7 +107,7 @@
               class="mr-2"
               small
               @click="banUser"
-            >Ban</v-btn>
+            >Mute</v-btn>
```
Comment (diingus) [inline components/Chat/ChatMessages/ChatMessageMenu.vue:110]: I still think 'master ignore list' is a better name. Might not be up to modern standards though

File: components/Chat/ChatSettings/ChatSettingsPanel/ChatBetaSettingsPanel.vue [hunk 2]
```diff
@@ -0,0 +1,82 @@
+<template>
+  <v-card
+    flat
+    tile
+    class="pb-3 px-3"
+  >
+    <!-- Beta Settings Subheader -->
+    <v-subheader class="overline mb-0">
```
Comment (diingus) [inline components/Chat/ChatSettings/ChatSettingsPanel/ChatBetaSettingsPanel.vue]:
```suggestion

```
No other tab starts with a subheader

File: components/Chat/ChatSettings/ChatSettingsPanel/ChatGeneralSettingsPanel.vue [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        hide-details
+        dense
+        inset
+      />
+
+      <!-- Receive @'s in Local -->
+      <v-switch
+        v-model="receiveMentionsInLocal"
+        :disabled="globalChat"
+        label="Recieve @'s while in local"
```
Comment (diingus) [inline components/Chat/ChatSettings/ChatSettingsPanel/ChatGeneralSettingsPanel.vue]: Spelling seems wierd :)

Comment (diingus) [reply components/Chat/ChatSettings/ChatSettingsPanel/ChatGeneralSettingsPanel.vue]: Apparently 'leave review' and 'leave single comment' do the same thing

File: components/Chat/ChatMessages/ChatMessageMenu.vue (hunk 1 above)
Comment (DispatchCommit) [reply components/Chat/ChatMessages/ChatMessageMenu.vue:110]: This wasn't supposed to sneak into this PR, will revisit later on.

File: components/Chat/ChatSettings/ChatSettingsPanel/ChatBetaSettingsPanel.vue (hunk 2 above)
Comment (DispatchCommit) [reply components/Chat/ChatSettings/ChatSettingsPanel/ChatBetaSettingsPanel.vue]: removed.

File: components/Chat/ChatSettings/ChatSettingsPanel/ChatGeneralSettingsPanel.vue (hunk 3 above)
Comment (DispatchCommit) [reply components/Chat/ChatSettings/ChatSettingsPanel/ChatGeneralSettingsPanel.vue]: fixed. In my defense, you made that label.

**Comment by DispatchCommit** [comment]
Updated preview
image

File: components/Chat/ChatSettings/ChatSettingsPanel/ChatTtsSettingsPanel.vue [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+      </v-alert>
+
+      <!-- TTS Voice Options -->
+      <div
+        v-if="ttsVoicesExist"
+      >
+        <!-- Enable TTS -->
+        <v-switch
+          v-model="useTts"
+          :disabled="!ttsVoicesExist"
```
Comment (diingus) [inline components/Chat/ChatSettings/ChatSettingsPanel/ChatTtsSettingsPanel.vue]: Seems redundant with the `v-if` on the same condition above.

File: components/Chat/ChatSettings/ChatSettingsPanel/ChatTtsSettingsPanel.vue [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+              class="mt-0 pt-0"
+              hide-details
+              single-line
+              type="number"
+              style="width: 50px"
+            />
+          </template>
+        </v-slider>
+
+        <i class="caption grey--text">note: a value of 0 disables timeout.</i>
```
Comment (diingus) [inline components/Chat/ChatSettings/ChatSettingsPanel/ChatTtsSettingsPanel.vue:131]: Maybe all settings should have a tooltip explaining them. Not sure what the mobile equivalent would be.

**Review by diingus — APPROVED** [review]
Looks good. There's a lot of boilerplate with `get()` and `set()` being mapped to store functions. Not sure if that could be factored out with clever use of the spread operator. 

Definitely a nice change, both in UX and in code, as it's no longer a mile long file with random containers keeping it all together.

File: components/Chat/ChatSettings/ChatSettingsPanel/ChatTtsSettingsPanel.vue (hunk 5 above)
Comment (DispatchCommit) [reply components/Chat/ChatSettings/ChatSettingsPanel/ChatTtsSettingsPanel.vue:131]: Should consider improving documentation all around tbh.

File: components/Chat/ChatSettings/ChatSettingsPanel/ChatTtsSettingsPanel.vue (hunk 4 above)
Comment (DispatchCommit) [reply components/Chat/ChatSettings/ChatSettingsPanel/ChatTtsSettingsPanel.vue]: true

File: components/Chat/ChatSettings/ChatSettingsPanel/ChatTtsSettingsPanel.vue (hunk 5 above)
Comment (diingus) [reply components/Chat/ChatSettings/ChatSettingsPanel/ChatTtsSettingsPanel.vue:131]: My `Chat.vue` diagram is just missing polls :)

Comment (DispatchCommit) [reply components/Chat/ChatSettings/ChatSettingsPanel/ChatTtsSettingsPanel.vue:131]: *dies inside*

**Pull request merged**